chiselFrontend/src/main/scala/chisel3/internal/Builder.scala

```diff
@@ -129,6 +133,11 @@ private[chisel3] trait HasId extends InstanceId {
     case Some(p) => p.name
     case None => throwException(s"$instanceName doesn't have a parent")
   }
+  // TODO Should this be public?
```

Comment:
For Testers, we do need an API to get the FIRRTL name of a Chisel object. How that should be done (through DataMirror maybe?) and specifics (name qualified up to which level? Top? User-defined root?) are questions though.